### PR TITLE
refactor: Feature dir structure

### PR DESCRIPTION
**Pull request by Sadhorsephile**
## Changes

- refactor feature dir structure according to Surf standarts
- refactor mason bricks
- actualize docs

## Checklist for self-check
- [x] Commits and PRs have been filed according to [the rules on the project](https://github.com/surfstudio/surf-flutter-app-template#workflow-in-a-repository).
- [x] The author is marked as an assigne and assigned mandatory reviewers.
- [x] Required labels marked
- [x] Specified related tasks and/or related PRs.
- [x] Specified Changes.
- [x] Attached videos/screenshots demonstrating the fix/feature.
- [x] All unspecified fields in the PR description deleted.
- [ ] New code covered by tests.

## Checklist for reviewers
- [ ] CI passed successfully _(with a green check mark)_.
- [ ] PR is atomic, by volume no more than 400 (+-) corrected lines (not including codogen).

Design:
- [ ] System design corresponds to the agreements on structure and architecture on the project.
- [ ] The code is decomposed into necessary and sufficient components.

Functionality:
- [ ] The code solves the problem.
- [ ] Any changes to the user interface are reasonable and look good.

Complexity:
- [ ] The code is clear, easy to read, functions are small, no more than 50 lines.
- [ ] The logic is not overcomplicated, there is no overengineering (no code sections that may be needed in the future, but no one knows about it).

Tests:
- [ ] Updated or added tests for mandatory components.
- [ ] The tests are correct, helpful, and well designed/developed.

Naming:
- [ ] The naming of variables, methods, classes and other components is understandable.

Comments:
- [ ] The comments are understandable and helpful.

Documentation:
- [ ] All labels are correct
- [ ] Technical documentation updated (after approval, updates last reviewer).

### CODE REVIEW

File: lib/features/info/presentation/screens/info/info_model.dart [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import 'package:elementary/elementary.dart';
+import 'package:flutter_template/features/info/presentation/screens/info/info_screen.dart';
+
+/// Model fol [InfoScreen].
+class InfoModel extends ElementaryModel {}
```
Comment (dzmitry-struk-surf) [inline lib/features/info/presentation/screens/info/info_model.dart:5]:
```suggestion
final class InfoModel extends ElementaryModel {}
```

File: lib/features/temp/data/repositories/temp_repository.dart [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import 'package:flutter_template/features/temp/domain/repositories/i_temp_repository.dart';
+
+/// Implementation of Template repository
+class TempRepository implements ITempRepository{}
```
Comment (dzmitry-struk-surf) [inline lib/features/temp/data/repositories/temp_repository.dart:4]:
```suggestion
final class TempRepository implements ITempRepository{}
```

File: lib/features/temp/di/temp_scope.dart [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+/// Interface of Template DI Scope.
+abstract interface class ITempScope implements IDisposableObject {
+  /// Template repository
+  abstract final ITempRepository templateRepository;
+}
+
+/// {@template temp_scope.class}
+/// Dependencies scope of Template module.
+/// {@endtemplate}
+class TempScope extends DisposableObject implements ITempScope {
```
Comment (dzmitry-struk-surf) [inline lib/features/temp/di/temp_scope.dart:15]:
```suggestion
final class TempScope extends DisposableObject implements ITempScope {
```

File: lib/features/temp/presentation/screens/temp/temp_model.dart [hunk 4]
```diff
@@ -1,27 +1,31 @@
 import 'package:elementary/elementary.dart';
 import 'package:flutter_template/config/environment/environment.dart';
 import 'package:flutter_template/features/common/service/theme/theme_service.dart';
-import 'package:flutter_template/features/temp/screens/temp_screen/temp_screen.dart';
+import 'package:flutter_template/features/temp/domain/repositories/i_temp_repository.dart';
+import 'package:flutter_template/features/temp/presentation/screens/temp/temp_screen.dart';
 
 /// Model for [TempScreen].
-class TempScreenModel extends ElementaryModel implements ITempScreenModel {
+class TempModel extends ElementaryModel implements ITempModel {
```
Comment (dzmitry-struk-surf) [inline lib/features/temp/presentation/screens/temp/temp_model.dart:8]:
```suggestion
final class TempModel extends ElementaryModel implements ITempModel {
```

File: lib/features/temp/presentation/screens/temp/temp_screen.dart [hunk 5]
```diff
@@ -1,22 +1,18 @@
...
-import 'package:flutter_template/features/navigation/domain/entity/app_route_names.dart';
-import 'package:flutter_template/features/temp/screens/temp_screen/temp_screen_widget_model.dart';
+import 'package:flutter_template/features/temp/presentation/screens/temp/temp_widget_model.dart';
 
 /// Initialization screens (this can be a HomeScreen or SplashScreen for example).
-@RoutePage(
-  name: AppRouteNames.tempScreen,
-)
-class TempScreen extends ElementaryWidget<TempScreenWidgetModel> {
+class TempScreen extends ElementaryWidget<TempWidgetModel> {
```
Comment (dzmitry-struk-surf) [inline lib/features/temp/presentation/screens/temp/temp_screen.dart:7]:
```suggestion
class TempScreen extends ElementaryWidget<ITempWidgetModel> {
```

File: lib/features/temp/presentation/screens/temp/temp_screen.dart [hunk 6]
```diff
@@ -1,22 +1,18 @@
...
   /// Create an instance [TempScreen].
   const TempScreen({
     Key? key,
     WidgetModelFactory wmFactory = initScreenWidgetModelFactory,
   }) : super(wmFactory, key: key);
 
   @override
-  Widget build(TempScreenWidgetModel wm) {
+  Widget build(TempWidgetModel wm) {
```
Comment (dzmitry-struk-surf) [inline lib/features/temp/presentation/screens/temp/temp_screen.dart:15]:
```suggestion
  Widget build(ITempWidgetModel wm) {
```

File: lib/features/temp/presentation/screens/temp/temp_widget_model.dart [hunk 7]
```diff
@@ -3,31 +3,34 @@ import 'package:elementary/elementary.dart';
...
 
 /// Widget model for [TempScreen].
-class TempScreenWidgetModel extends WidgetModel<TempScreen, ITempScreenModel>
+class TempWidgetModel extends WidgetModel<TempScreen, ITempModel>
     with ThemeWMMixin
-    implements IDebugWidgetModel {
+    implements ITempWidgetModel {
   final _defaultNavBarItems = [
     const BottomNavigationBarItem(
       label: 'Dash screen',
```
Comment (dzmitry-struk-surf) [inline lib/features/temp/presentation/screens/temp/temp_widget_model.dart:36]: This should be in localization.

File: lib/features/temp/presentation/temp_flow.dart [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import 'package:flutter_template/features/common/widgets/di_scope/di_scope.dart';
+import 'package:flutter_template/features/navigation/domain/entity/app_route_names.dart';
+import 'package:flutter_template/features/temp/di/temp_scope.dart';
+import 'package:flutter_template/features/temp/presentation/screens/temp/temp_screen.dart';
+
+/// Entry of template feature.
+@RoutePage(name: AppRouteNames.tempScreen)
+class TempFlow extends StatelessWidget implements AutoRouteWrapper {
+  /// {@macro scan_biomarkers_entry.class}
+  const TempFlow({super.key});
```
Comment (dzmitry-struk-surf) [inline lib/features/temp/presentation/temp_flow.dart:12]: You should not use `@macro` without previously using the `@template`. 
Add `@template` and change `@macro` for the relevant class, not `scan_biomarkers_entry.class`.

**Review by dzmitry-struk-surf — COMMENTED** [review]
Add `const` constructors for the repositories where it is possible, please.

File: docs/architecture_and_project_structure.md [hunk 9]
```diff
@@ -34,16 +34,18 @@
...
-                - {screen_name}
-                    - widget
-                    - wm
-                    - model
-            - widgets
+                - entities
+                - repositories *(interfaces)*
+            - data
+              - repositories *(implementations)*
+              - mappers
```
[the remark this answers is not in the document]
Comment (Evgenia-bit) [reply docs/architecture_and_project_structure.md:41]: you mean swap `data` and `domain`?

File: docs/architecture_and_project_structure.md [hunk 10]
```diff
@@ -34,16 +34,18 @@
...
+            - presentation
+                - screens
+                    - {screen_name}
+                        - widget
+                        - wm
+                        - model
```
[the remark this answers is not in the document]
Comment (Evgenia-bit) [reply docs/architecture_and_project_structure.md:47]: do you really need to add it here, and not like this?:
Screenshot 2024-02-19 at 11 20 59

File: docs/architecture_and_project_structure.md [hunk 11]
```diff
@@ -34,16 +34,24 @@
...
+            - presentation
+              if feature contains multiple sub-features/screens
+                - {sub_feature_name/screen_name}
+                    - {sub_feature_name/screen_name}_widget.dart
+                    - {sub_feature_name/screen_name}_wm.dart
+                    - {sub_feature_name/screen_name}_model.dart
+              else 
+                - {sub_feature_name/screen_name}_widget.dart
+                - {sub_feature_name/screen_name}_wm.dart
+                - {sub_feature_name/screen_name}_model.dart
```
Comment (dzmitry-struk-surf) [inline docs/architecture_and_project_structure.md:51]: Here it should be just:
```suggestion
                - {feature_name}_widget.dart
                - {feature_name}_wm.dart
                - {feature_name}_model.dart
```

File: docs/architecture_and_project_structure.md [hunk 12]
```diff
@@ -34,16 +34,24 @@
...
+
+                - widgets
+                {feature_name}_flow.dart
```
Comment (dzmitry-struk-surf) [inline docs/architecture_and_project_structure.md:54]: `{feature_name}_flow.dart` is an entry point for the feature (or sub_feature) so it should be here in `else` case and inside sub-feature's folder when we have a sub-feature.

**Review by dzmitry-struk-surf — CHANGES_REQUESTED** [review]

File: docs/architecture_and_project_structure.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (dzmitry-struk-surf) [reply docs/architecture_and_project_structure.md:41]: Yep, sorry.

File: docs/architecture_and_project_structure.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (dzmitry-struk-surf) [reply docs/architecture_and_project_structure.md]: `flow` is an entry point for the feature (or sub-feature).

File: docs/architecture_and_project_structure.md [hunk 13]
```diff
@@ -88,19 +102,33 @@ The folder has the code of the project and consists of the following folders:
 
   The structure of features:
 
+
   - di — dependency injection containers.
+  - data - contains:
+    - converters - classes for mapping data from DTO in entity and vice versa. These classes extends the abstract class `Converter`.
```
Comment (dzmitry-struk-surf) [inline docs/architecture_and_project_structure.md:108]:
```suggestion
    - converters - classes for mapping data from DTO in entity and vice versa. These classes extend the abstract class `Converter`.
```

File: docs/architecture_and_project_structure.md [hunk 14]
```diff
@@ -88,19 +102,33 @@ The folder has the code of the project and consists of the following folders:
...
-      - wm - WidgetModel.
-      - model - ElementaryModel.
-  - widgets — widgets relevant to the feature.
+    - entities — business data models.
+    - repositories — interfaces for repositories relevant to the feature.
+  - presentation - contains:
+
+   **if feature contains multiple sub-features/screens**
+
+    - {sub_feature_name/screen_name} - screens or sub-features relevant to the feature, with each screen in a separate self-titled folder containing the following in separate files:
```
Comment (dzmitry-struk-surf) [inline docs/architecture_and_project_structure.md:117]:
```suggestion
    - {sub_feature_name/screen_name} - screens or sub-features relevant to the feature, with each screen or sub-feature in a separate self-titled folder containing the following in separate files:
```

**Review by dzmitry-struk-surf — CHANGES_REQUESTED** [review]

**Review by dzmitry-struk-surf — APPROVED** [review]

**Pull request merged**